rust/lance-table/src/rowids.rs

```diff
@@ -206,6 +212,215 @@ impl RowIdSequence {
 
         (row_ids, segment_ranges)
     }
+
+    pub fn slice(&self, offset: usize, len: usize) -> RowIdSeqSlice<'_> {
+        // Find the starting position
+        let mut offset_start = offset;
+        let mut segment_offset = 0;
+        for segment in &self.0 {
+            let segment_len = segment.len();
+            if offset_start < segment_len {
+                break;
+            }
+            offset_start -= segment_len;
+            segment_offset += 1;
+        }
+
+        // Find the ending position
+        let mut offset_last = offset_start + len;
+        let mut segment_offset_last = segment_offset;
+        for segment in &self.0[segment_offset..] {
+            let segment_len = segment.len();
+            if offset_last <= segment_len {
+                break;
+            }
+            offset_last -= segment_len;
+            segment_offset_last += 1;
+        }
+
+        RowIdSeqSlice {
+            segments: &self.0[segment_offset..=segment_offset_last],
+            offset_start,
+            offset_last,
+        }
+    }
+
+    pub fn get(&self, index: usize) -> Option<u64> {
```

Comment:
Does this only return None if `index` is greater than the size of the sequence?

Reply:
Yes. Documented that.